checks.d/storm_rest_api.py

```diff
@@ -0,0 +1,301 @@
+                   tags=config.tags)
+        self.gauge(self.metric(config, 'storm.rest.cluster.executors_total_count'),
+                   cluster['executorsTotal'],
+                   tags=config.tags)
+        self.gauge(self.metric(config, 'storm.rest.cluster.tasks_total_count'),
+                   cluster['tasksTotal'],
+                   tags=config.tags)
+
+
+    def report_supervisors(self, config, resp):
+        self.gauge(self.metric(config, 'storm.rest.supervisors_total'),
+                   len(resp.get('supervisors', [])),
+                   tags=config.tags)
+        for supe in resp.get('supervisors', []):
+            supe_tags = [
+                'host:' + supe.get('host'),
+            ] + config.tags
+            self.gauge(self.metric(config, 'storm.rest.supervisor.slots_total'),
+                       supe.get('slotsTotal'), tags=supe_tags)
+            self.gauge(self.metric(config, 'storm.rest.supervisor.slots_used_total'),
+                       supe.get('slotsUsed'), tags=supe_tags)
+            self.monotonic_count(self.metric(config, 'storm.rest.supervisor.uptime_seconds'),
+                                 storm_utils.translate_timespec(supe.get('uptime')), tags=supe_tags)
+
+    def _topology_name(self, config, topology):
+        """Returns the name if a config's topology regex matches, None otherwise."""
+        match = config.topologies.match(topology.get('name'))
+        if match:
+            return match.group(1)
+        else:
+            return None
+
+    def report_topologies(self, config, topologies):
+        """Report metadata about topologies that match the topologies regex.
+        """
+        for pretty_name, topo in topologies.iteritems():
+            uptime = topo.get('uptime', '0s')
+            tags = config.tags + [
+                'storm_topology:' + pretty_name,
+            ]
+            self.monotonic_count(self.metric(config, 'storm.rest.topologies.uptime_seconds'),
+                                 storm_utils.translate_timespec(uptime), tags=tags)
+            self.gauge(self.metric(config, 'storm.rest.topologies.tasks_total'),
+                       topo['tasksTotal'], tags=tags)
+            self.gauge(self.metric(config, 'storm.rest.topologies.workers_total'),
+                       topo['workersTotal'], tags=tags)
+            self.gauge(self.metric(config, 'storm.rest.topologies.executors_total'), topo['executorsTotal'], tags=tags)
+
+    def task_id_tags(self, config, component_type, task_id):
+        return config.task_tags.get(component_type, {}).get(task_id, [])
+
+    def report_topology(self, config, name, details):
+        """
+        Report statistics for a single topology's spouts and bolts.
+        """
+        tags = config.tags + [
+            'storm_topology:' + name,
+        ]
+        def report_task(component_type, name, task, tags):
+            self.gauge(self.metric(config, ('storm.rest.%s.executors_total' % component_type)),
```

Comment:
Can we find a better suffix than total here? Since this isn't a counter the `_total` will be misleading with the current [metric naming standards](https://hackpad.corp.stripe.com/Introduction-to-Datadogs-Data-Model-pNJvI5fuUxG#:h=Standards-for-Metric-Naming-an). Gauges are hard to name. 😿